Initium-Odp/src/com/universeprojects/miniup/server/model/GridCell.java

```diff
@@ -1,9 +1,18 @@
 package com.universeprojects.miniup.server.model;
 
-public class GridBackground {
+public class GridCell {
 	private String backgroundFile;
+	private int xGridCoord;
```

Comment:
It would be less work for updates if you just stored everything in a map to begin with. You could even store it in a JSONObject so that it's ready to go. Especially since you're manually writing a json string in the toString(). The only downside is having to cast in your getters.